wordfugue/blog/models.py

```diff
@@ -1,10 +1,96 @@
-from django.db import models
 from django.conf import settings
-from django.urls import reverse
 from django.contrib.sites.models import Site
+from django.db import models
+from django.urls import reverse
 from django.utils import timezone
 from django.utils.html import strip_tags
 from model_utils.models import TimeStampedModel
+from modelcluster.contrib.taggit import ClusterTaggableManager
+from modelcluster.fields import ParentalKey
+from taggit.models import TaggedItemBase
+from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, MultiFieldPanel
+from wagtail.core.fields import RichTextField
+from wagtail.core.models import Orderable, Page
+from wagtail.images.edit_handlers import ImageChooserPanel
+from wagtail.search import index
+
+
+class BlogIndexPage(Page, TimeStampedModel):
+    intro = RichTextField(blank=True)
+
+    content_panels = Page.content_panels + [FieldPanel("intro", classname="full")]
+
+    def __str__(self):
+        return f"Blog Index: {self.title}"
+
+    def get_admin_display_title(self):
+        return str(self)
+
+    def get_context(self, *args, **kwargs):
+        context = super().get_context(*args, **kwargs)
+
+        context["blog_entries"] = (
+            BlogPage.objects.child_of(self).live().order_by("-date")
```

Comment:
If I didn't do this, wagtail would render the blog post list in the reverse order of how I wanted.